### ISSUE TITLE
random-data-generator causes infinitely loop

### ISSUE THREAD

**Issue by pandaforme** [DanielaSfregola/random-data-generator#33]
The codes will cause infinitely loop:

```
import com.danielasfregola.randomdatagenerator.RandomDataGenerator
import org.scalatest.prop.PropertyChecks
import org.scalatest.{Matchers, WordSpec}

class Test extends WordSpec with Matchers with PropertyChecks with RandomDataGenerator {
  sealed trait Test1 {
    val value: Int
  }

  object Test1 {
    def toStatus(i: Int): Test1 = i match {
      case AA.value => AA
      case BB.value => BB
      case _ => CC
    }

    def toString(s: Test1): String = s match {
      case AA => "Unverified"
      case BB => "Verified"
      case CC => "dd"
    }

    case object AA extends Test1 {
      override val value: Int = 0
    }

    case object BB extends Test1 {
      override val value: Int = 1
    }

    case object CC extends Test1 {
      override val value: Int = -1
    }
  }

  "OK" in {
    import Test1._
    forAll { x: Test1 =>
      x shouldBe an[AA.type]
    }
  }
}
```

But when I removed one instance from Test1, for example, CC
It works well and doesn't run infinitely...

```
import com.danielasfregola.randomdatagenerator.RandomDataGenerator
import org.scalatest.prop.PropertyChecks
import org.scalatest.{Matchers, WordSpec}

class Test extends WordSpec with Matchers with PropertyChecks with RandomDataGenerator {
  sealed trait Test1 {
    val value: Int
  }

  object Test1 {
    def toStatus(i: Int): Test1 = i match {
      case AA.value => AA
      case BB.value => BB
    }

    def toString(s: Test1): String = s match {
      case AA => "Unverified"
      case BB => "Verified"
    }

    case object AA extends Test1 {
      override val value: Int = 0
    }

    case object BB extends Test1 {
      override val value: Int = 1
    }
  }

  "OK" in {
    import Test1._
    forAll { x: Test1 =>
      x shouldBe an[AA.type]
    }
  }
}
```

Is it a bug or I misused it?

**Comment by DanielaSfregola** [comment]
Thanks for raising this @pandaforme! I'll look into it asap

**Comment by DanielaSfregola** [comment]
Hi @pandaforme,
I managed to replicate the bug that is caused by a dependency that we use that is called `scalacheck-shapeless`. See https://github.com/alexarchambault/scalacheck-shapeless/issues/55 for more information on the bug.

You are not actually using `RandomDataGenerator` as you are not calling the `random[T]` function to generate a random sample of `T`.

However, by extending `RandomDataGenerator` you are adding [scalacheck-shapeless](https://github.com/alexarchambault/scalacheck-shapeless) to your scope. `scalacheck-shapeless` is used to derivate an instance of Arbitrary[Test1], but unfortunately, it never terminates if there are more than two case objects.

While waiting for a fix in `scalacheck-shapeless` you have the workaround of manually generate the arbitrary.  See for example the following (simplified from your example) snapshot of code:

```scala
import org.scalacheck.{Arbitrary, Gen}
import org.scalatest.prop.PropertyChecks
import org.scalatest.{Matchers, WordSpec}

class Test extends WordSpec with Matchers with PropertyChecks {

  sealed trait Test1

  case object AA extends Test1
  case object BB extends Test1
  case object CC extends Test1

  implicit val arb: Arbitrary[Test1] = Arbitrary(Gen.oneOf(AA, BB, CC))

  "OK" in {

    forAll { x: Test1 =>
      x shouldBe an[AA.type]
    }
  }
}
```
Note, no need to extend `RandomDataGenerator` as you are not using the `random[T]` function.

Cheers,
Daniela

**Comment by DanielaSfregola** [comment]
This should now be fixed in version 2.2
